### DISCUSSION TITLE
How to mock backend calls [Spectacular]

### DISCUSSION THREAD

**Discussion by meirka** [ngworker/ngworker#32] [Q&A] [answered]
How do I mock backend calls in Spectacular? HttpTestingController (@angular/common/http/testing ) ?

I don't want to mock services, I prefer using HttpTestingController (in case of angular), WireMock (in case of Java) or Nock (in case of Nodejs backend like NestJS).

Thank you!

**Comment by LayZeeDK** [comment, marked as answer]
Yes, you can use Angular's [HttpClientTestingModule](https://angular.io/api/common/http/testing/HttpClientTestingModule), [provideHttpClientTesting](https://angular.io/api/common/http/testing/provideHttpClientTesting), [angular-in-memory-web-api](https://www.npmjs.com/package/angular-in-memory-web-api), or Mock Service Worker\[[1](https://github.com/mswjs/examples/tree/master/examples/rest-angular)\]\[[2](https://timdeschryver.dev/blog/using-msw-in-an-angular-project)\].
